library/core/src/pin.rs

```diff
@@ -406,7 +406,9 @@ use crate::ops::{CoerceUnsized, Deref, DerefMut, DispatchFromDyn, Receiver};
 #[repr(transparent)]
 #[derive(Copy, Clone)]
 pub struct Pin<P> {
-    pointer: P,
+    #[unstable(feature = "unsafe_pin_internals", issue = "none")]
+    #[doc(hidden)]
+    pub pointer: P,
```

Comment:
Might be worth adding a comment here that explains why this is `pub`, and the hack that seems to make this still (somewhat) sound.

Reply:
Addressed in https://github.com/rust-lang/rust/pull/93176/commits/002f627d383597a792afa14a8b9046659dabdaa2